packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -134,34 +131,18 @@ const handleResource = (resourceName, context, props) => {
     cacheKey = JSON.stringify({ resourceName, ...props, mode })
   }
 
-  // update global context when results come in.
-  const updateResource = result => {
-    allResources = allResources.filter(a => a.resourceDefinitions._key)
-    const resourceMap = new Map()
-
-    allResources.forEach(r => resourceMap.set(r.resourceDefinitions._key, r))
-    const newResource = {
-      resourceName,
-      resourceDefinitions: props,
-      ...result,
-    }
-
-    if (!lodash.isEqual(newResource, resourceMap.get(trueKey))) {
-      resourceMap.set(trueKey, newResource)
-      // TODO Do we need this? It's causing infinite loops
-      // setResources([...resourceMap.values()])
-    }
-  }
-
   let allResources = useResourceContext()
+  const resourcePlan = allResources?.find(
```

Comment:
Nit: I would put this new block above the other new block (currently line 162) since `resourcePlan` is not used before that point and code might return early.

Reply:
yeah good catch — I'd been thinking of using this elsewhere but no reason to run unless it's actually being used.